dfsclient/src/main/java/org/hpccsystems/dfs/client/AvroFieldTranslator.java

```diff
@@ -0,0 +1,1036 @@
+                    ret = byteBuffer.array();
+                }
+                else if (fieldMeta != null)
+                {
+                    return fieldMeta.defaultVal();
+                }
+
+                return ret;
+            }
+            case DOUBLE:
+            {
+                if (fd.getFieldType() != FieldType.REAL)
+                {
+                    throw new Exception("AvroFieldTranslator toHPCC: incompatible avro schema type: " 
+                                        + schema.getType() + "  to hpcc field type: " + fd.getFieldType());
+                }
+
+                Double ret = DEFAULT_DOUBLE;
+                if (field != null)
+                {
+                    ret = (Double) field;
+                }
+                else if (fieldMeta != null)
+                {
+                    return fieldMeta.defaultVal();
+                }
+
+                return ret;
+            }
+            case ENUM:
+            {
+                if (fd.getFieldType() != FieldType.INTEGER)
+                {
+                    throw new Exception("AvroFieldTranslator toHPCC: incompatible avro schema type: " 
+                                        + schema.getType() + "  to hpcc field type: " + fd.getFieldType());
+                }
+
+                Integer ret = -1;
+                if (field != null)
+                {
+                    GenericEnumSymbol enumSymbol = (GenericEnumSymbol) field;
+                    ret = schema.getEnumOrdinal(enumSymbol.toString());
+                }
+                else if (fieldMeta != null)
+                {
+                    return fieldMeta.defaultVal();
+                }
+
+                return ret;
+            }
+            case FIXED:
+            {
+                if (schema.getLogicalType() instanceof LogicalTypes.Decimal)
+                {
+                    if (fd.getFieldType() != FieldType.DECIMAL)
+                    {
+                        throw new Exception("AvroFieldTranslator toHPCC: incompatible avro schema type: " 
+                                            + schema.getLogicalType() + "  to hpcc field type: " + fd.getFieldType());
+                    }
+
+                    // Fixed binary represents a big-endian encoded integer.
+                    // BigInt -> Decimal conversion *= 10^-Scale
+                    LogicalTypes.Decimal decimalInfo = (LogicalTypes.Decimal) schema.getLogicalType();
+
+                    BigDecimal ret = BigDecimal.ZERO;
+                    if (field != null)
+                    {
+                        byte[] data = ((GenericFixed) field).bytes();
+                        BigInteger intValue = new BigInteger(data);
+
+                        ret = new BigDecimal(intValue,decimalInfo.getScale());
+                    }
+
+                    return ret;
+                }
+                else if (schema.getLogicalType().getName().equals("duration"))
```

Comment:
minor, but let's try to avoid the stringwise comparison if possible, is there a logicaltype code?

Reply:
There doesn't appear to be a better way to compare at the moment.